onadata/apps/api/tests/viewsets/test_floip_viewset.py

```diff
@@ -196,6 +207,50 @@ def test_publish_number_question_names(self):  # pylint: disable=C0103
             self.assertIn(u"The name '1448506769745_42' is an invalid xml tag",
                           response.data['text'])
 
+    def test_retrieve_responses_merged_dataset(self):
+        """
+        Test retrieving responses from the FLOIP endpoint
+        """
+        merged_dataset_view = MergedXFormViewSet.as_view({
+            'post': 'create',
+        })
+
+        self.project = get_user_default_project(self.user)
+        self._publish_xls_form_to_project()
+        self._make_submissions()
+        xform = self._publish_markdown(MD, self.user, id_string='a')
+
+        data = {
+            'xforms': [
+                f'http://testserver/api/v1/forms/{self.xform.pk}',
+                f'http://testserver/api/v1/forms/{xform.pk}',
+            ],
+            'name':
+            'Merged Dataset',
+            'project':
+            f'http://testserver/api/v1/projects/{self.project.pk}',
+        }
+
+        request = self.factory.post('/', data=data, **self.extra)
+        response = merged_dataset_view(request)
+        self.assertEqual(response.status_code, 201)
+        dataset_uuid = response.data['uuid']
+
+        # Assert that it's possible to retrieve the responses
+        view = FloipViewSet.as_view({'get': 'responses'})
+        request = self.factory.get(
+            f'/flow-results/packages/{dataset_uuid}/responses',
+            content_type='application/vnd.api+json', **self.extra)
+        response = view(request, uuid=dataset_uuid)
+        self.assertEqual(response.status_code, 200)
+
+        # Convert the returned generator object into a list
+        response.data['attributes']['responses'] = list(
+            response.data['attributes']['responses'])
+        # The transportation form(self.xform) contains 11 responses
+        # Assert that the responses are returned
+        self.assertEqual(len(response.data['attributes']['responses']), 11)
```

Comment:
Chose to use the number of responses returned(11) for this test due to erratic time values returned in the responses.